src/bootstrap.jsx

```diff
@@ -155,8 +155,12 @@ function initializeApi(store) {
 
 function initRootRoute({ store, routes, indexRedirect, menuItems, redirects }) {
   const setActiveModule = (module) => {
-    const action = actions.modules.setActive(module);
-    store.dispatch(action);
+    // active module for these two will be handled by components themselves
+    // because in some situations workspace should be chosen instead of study-manager
+    if (!['study-manager', 'analysis-execution'].includes(module)) {
```

Comment:
Looks not cool that generic function know smth about its param. Why do you need this condition - why just not to leave everything like it was before?